### PR TITLE
Job update

### LINKED ISSUE #14308: Experimental: Job status not being updated
We need code in the job controller to update the status of the job.  That isn't happening right now.

@mikedanese @soltysh

### PR DESCRIPTION
**Pull request by erictune**
Fixes  #14308

### CODE REVIEW

**Comment by erictune** [comment]
@soltysh @mikedanese

**Comment by soltysh** [comment]
LGTM, shippable is failing recently more than expected imho.

**Comment by mikedanese** [comment]
LGTM

File: pkg/registry/job/strategy.go [hunk 1]
```diff
@@ -79,6 +79,22 @@ func (jobStrategy) ValidateUpdate(ctx api.Context, obj, old runtime.Object) fiel
...
+var StatusStrategy = jobStatusStrategy{Strategy}
+
+func (jobStatusStrategy) PrepareForUpdate(obj, old runtime.Object) {
+	newJob := obj.(*experimental.Job)
+	oldJob := old.(*experimental.Job)
+	newJob.Spec = oldJob.Spec
+}
+
+func (jobStatusStrategy) ValidateUpdate(ctx api.Context, obj, old runtime.Object) fielderrors.ValidationErrorList {
+	return validation.ValidateJobUpdateStatus(obj.(*experimental.Job), old.(*experimental.Job))
```
Comment (derekwaynecarr) [inline pkg/registry/job/strategy.go]: @erictune - I missed this in initial review, but this appears to be flipped with the validation call that takes (old, new) and not (new, old)

**Pull request merged**